.azure-pipelines/pipelines.yml

```diff
@@ -37,6 +37,15 @@ jobs:
           artifactName: format
         condition: failed()
 
+  - job: examples
+    dependsOn: [] # this removes the implicit dependency on previous stage and causes this to run in parallel.
+    pool:
+      vmImage: "ubuntu-18.04"
+    steps:
+      - script: ci/verify_examples.sh
```

Comment:
I'm OK do this as part of existing docker job, so you don't have to wait for the image built.

Reply:
im not totally following here, couple of questions

- whereabouts in ci is best to run this job ?
- are you happy with running these tests with `docker-in-docker` ?

Reply:
i guess theres another question about the envoy container that is being tested in the examples - atm it would use the published one - ie not the one which would be built from the tested branch

Reply:
I'm saying run this as a part of https://github.com/envoyproxy/envoy/blob/master/ci/docker_ci.sh

We used to have a TODO for that and lost during refactoring: https://github.com/envoyproxy/envoy/blob/v1.13.0/ci/docker_push.sh#L24-L27

Reply:
Also I'm not sure if we need docker-in-docker, just use docker seems fine?

Reply:
for ci - mostly yep it doesnt matter - although it is quite handy for the env to be clean to test the setup - ie user/perms/umask etc

running locally it matters if you are running docker already and dont want lots of new images and/or the script to prune anything you currently have in your docker env

Reply:
hmm

actually it does matter specifically for the umask part of the test

if the repo is already checked out and the umask is not set then these tests would pass without really testing the permissions issue

we could get around that by settting the umask before checking out the repo